packages/visual-editor/src/hooks/useDraggablePosition.ts

```diff
@@ -0,0 +1,38 @@
+import { useDraggedItemStore } from '@/store/draggedItem';
+import { DraggablePosition } from '@/types/constants';
+import { MutableRefObject, useEffect } from 'react';
+
+interface Params {
+  draggableRef: MutableRefObject<HTMLElement | null | undefined>;
```

Comment:
I think this type can be changed to `MutableRefObject<HTMLElement | null>` as there doesn't seem to be a case where the value would be `undefined`.

The `useRef` in `<DraggableComponent>` and `<DraggableComponentList>` would need to be changed to also use `<HTMLElement | null>` to match

Reply:
Good catch!